src/app/code/community/Shopgate/Cloudapi/Model/Api2/Customers/Rest.php

```diff
@@ -22,4 +22,47 @@
 
 class Shopgate_Cloudapi_Model_Api2_Customers_Rest extends Shopgate_Cloudapi_Model_Api2_Resource
 {
+    /**
+     * Get customer resource validator instance
+     *
+     * @return Shopgate_Cloudapi_Model_Api2_Customers_Validator
+     */
+    protected function _getValidator()
+    {
+        /** @noinspection PhpIncompatibleReturnTypeInspection */
+        return Mage::getModel('shopgate_cloudapi/api2_customers_validator', array('resource' => $this));
+    }
+
+    /**
+     * Sets detailed validation errors to be returned by the address endpoints
+     *
+     * @param Shopgate_Cloudapi_Model_Api2_Customers_Validator $validator
+     *
+     * @return Shopgate_Cloudapi_Model_Api2_Customers_Rest
+     */
+    protected function setDetailedErrors( $validator)
```

Comment:
I think instead of copy/pasting, you should move this logic in to the **main** validator class and adjust the `address` class usage too. Addresses were not released yet, so can be cleaned up without backwards compatibility issues.